web/src/main/java/org/openmrs/web/filter/initialization/InitializationFilter.java

```diff
@@ -1690,6 +1691,13 @@ public void executing(ChangeSet changeSet, int numChangeSetsToRun) {
 						setExecutingTask(null);
 						setMessage("Starting OpenMRS");
 						
+						// Update the filter to set the Openmrs classloader on the current thread before
+						// starting OpenMRS
+						// and the spring web application context that way it is inherited by the daemon
+						// thread.
+						Thread.currentThread().setContextClassLoader(OpenmrsClassLoader.getInstance());
```

Comment:
Why on this line but not before?

Reply:
the reason i put it on this line is so that the` OpenmrsClassLoader` is immediately loaded just before spring is booted  next below with `contextLoader.initWebApplicationContext(filterConfig.getServletContext());`